### ISSUE TITLE
Consider providing `pkg-config` files (especially for `eigen3`)

### ISSUE THREAD

**Issue by EricCousineau-TRI** [RobotLocomotion/drake#7355]
Since some incantations of `FindEigen3.cmake` use `pkg-config`, we may want to consider packaging `pkg-config` files using Matt's CPS setup, at least for upstream dependencies.

If possible, it'd be nice to have a `pkg-config` file for `drake` as well, but I would not see that as critical.

**Comment by jamiesnape** [comment]
I imagine we will be switching to the system version of Eigen3 at some point. For the rest of Drake, we made a conscience decision not to support `pkg-config`. Obviously that could be revisited, but supporting both Bazel and CMake is already a lot of work.

**Comment by jwnimmer-tri** [comment]
> I imagine we will be switching to the system version of Eigen3 at some point.

Uncertain, but my guess would be not until Ubuntu 18.04 Bionic.

**Comment by jwnimmer-tri** [comment]
FYI We will _not_ use Eigen3 from Bionic; it's not `>= 3.3.3`, which is our baseline.

**Comment by jwnimmer-tri** [comment]
Actually, Bionic has `3.3.4` now.

**Comment by jwnimmer-tri** [comment]
Oops, I somehow fat-fingered the assignee widget yesterday.

**Comment by jwnimmer-tri** [comment]
@EricCousineau-TRI now that we use the pre-existing Eigen (not installing our own), I think this issue can be closed?

**Comment by EricCousineau-TRI** [comment]
Yup!

(Also, this was written for `catkin` / ROS1-based workflows; given that ROS2 / `ament_cmake` no longer seem to generate / interface with `pkg-config`, yeah, it's way less important.)
